core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java

```diff
@@ -202,8 +203,17 @@ public static PoolingDataSource setupPooledDataSource( IDatabaseConnection datab
        * ConnectionFactory creates connections on behalf of the pool. Here, we use the DriverManagerConnectionFactory
        * because that essentially uses DriverManager as the source of connections.
        */
-      ConnectionFactory factory =
-          new DriverManagerConnectionFactory( url, databaseConnection.getUsername(), databaseConnection.getPassword() );
+      ConnectionFactory factory = null;
+      if( url.contains( "mysql" ) ) {
+        Properties props = new Properties();
+        props.put( "user", databaseConnection.getUsername() );
+        props.put( "password", databaseConnection.getPassword() );
+        props.put( "socketTimeout", "0" );
+        props.put( "connectTimeout", "100" );
```

Comment:
Hey I'm "okay" with the conditional branch, but this connectTimeout is way too low. Please increase this to something reasonable like 5000.